router/core/request_context_fields.go

```diff
@@ -44,7 +45,7 @@ func NewExpressionLogField(val any, key string, defaultValue any) zap.Field {
 		val = &assertVal
 	}
 
-	if v := val; v != "" {
+	if v := val; v != "" && v != nil {
```

Comment:
While we validate no explicit nil can be returned, on case we use the optional chaining operator we can still get nil, which will result in no default value being set.
(This could be considered a bit of a breaking change, if nil is returned somehow we don't set the default value right now.)